### PR TITLE
feat: add mremap() via sallyport update

### PR DESCRIPTION
**Pull request by haraldh**
simple version with copying to a new location

Fixes: #1638
Fixes: #1637

Signed-off-by: Harald Hoyer <harald@profian.com>

### CODE REVIEW

**Review by rvolosatovs — COMMENTED** [review]
Rather than maintaining 2 identical implementations for 2 different shims, why not implement this within sallyport directly?

**Comment by haraldh** [comment]
> Rather than maintaining 2 identical implementations for 2 different shims, why not implement this within sallyport directly (so shims can override the default implementation if required)?

This will be implemented in a specialized version in the end anyway.

**Comment by haraldh** [comment]
but yeah, let's do it in sallyport

File: internal/shim-kvm/src/hostcall.rs [hunk 1]
```diff
@@ -439,6 +441,80 @@ impl Handler for HostCall<'_> {
...
+                            );
+
+                            eprintln!("SC> mremap() shrink = {:?}", target_addr);
+
+                            target_addr
+                        }
+                        Ordering::Less => {
+                            // simply copy the old data to a new location
+
+                            // FIXME: find out the permissions of the old segment
```
Comment (jarkkojs) [inline internal/shim-kvm/src/hostcall.rs:488]: `mmledger` could be used for this. it does not have query function for this but nothing prevents to add one.

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**